torch/_decomp/decompositions.py

```diff
@@ -3434,6 +3445,54 @@ def arange_start(
     )
 
 
+def linalg_solve_is_vector_rhs(input: Tensor, other: Tensor) -> bool:
+    expected_batched_rhs_shape = input.shape[:-1]
+    vector_case = other.ndim == 1 or (
+        input.ndim - 1 == other.ndim and other.shape == expected_batched_rhs_shape
+    )
+    return vector_case
+
+
+@register_decomposition([aten._linalg_solve_ex.default, aten._linalg_solve_ex.result])
+@out_wrapper("result", "LU", "pivots", "info")
+def _linalg_solve_ex(
+    A: Tensor, B: Tensor, *, left: bool = True, check_errors: bool = False
+) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
+    checkFloatingOrComplex(A, "linalg.solve")
+    utils.check(
+        A.dtype == B.dtype,
+        lambda: (
+            f"linalg.solve: Expected A and B to have the same dtype, but found A of type "
+            f"{A.dtype} and B of type {B.dtype} instead"
+        ),
+    )
+    vector_case = linalg_solve_is_vector_rhs(A, B)
+    B_ = B.unsqueeze(-1) if vector_case else B
+    checkInputsSolver(A, B_, left, "linalg.solve")
+    B_broad_shape, _ = _linalg_broadcast_batch_dims(B_, A)
+    utils.check(
+        left or not vector_case,
+        lambda: (
+            "linalg.solve: Vector broadcasting of the left hand side is not supported for left=False. "
+            "In this case linalg.solve is equivalent to B / A.squeeze(-1)"
+        ),
+    )
+    use_A_T = A.is_contiguous() and not A.is_complex()
+    LU, pivots, info = torch.linalg.lu_factor_ex(A.mT if use_A_T else A)
+    if check_errors:
+        aten._linalg_check_errors(info, "torch.linalg.solve_ex", is_matrix=A.ndim == 2)
```

Comment:
Is it okay to call aten ops from here? I've seen it done somewhere. Not sure when it's appropriate and when not.

Reply:
Yes it's fine to call aten ops.